shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -223,6 +226,82 @@ class JSBackend {
     )
   }
 
+  protected def translateTraitDeclaration(
+      traitSymbol: TraitSymbol
+  )(implicit scope: Scope): JSConstDecl = {
+    import JSCodeHelpers._
+    val stmts = traitSymbol.methods.map { method =>
+      val name = method.nme.name
+      method.rhs.value match {
+        case Lam(params, body) =>
+          val methodParams = translateParams(params)
+          val methodScope = scope.derive(s"Method $name", JSPattern.bindings(methodParams))
+          methodScope.declareValue("this")
+          id("instance")(name) := JSFuncExpr(
+            N,
+            Nil,
+            `return`(translateTerm(body)(methodScope)) :: Nil
+          )
+        case term =>
+          val getterScope = scope.derive(s"Getter $name")
+          getterScope.declareValue("this")
+          id("Object")("defineProperty")(
+            id("instance"),
+            JSExpr(name),
+            JSRecord(
+              "enumerable" -> JSLit("true") ::
+              "get" -> JSFuncExpr(
+                N,
+                Nil,
+                `return`(translateTerm(term)(getterScope)) :: Nil
+              ) :: Nil
+            )
+          ).stmt
+      }
+    }
+    val implement = JSFuncExpr(
+      S("implement"),
+      param("instance") :: Nil,
+      id("Object")("defineProperty")(
+        id("instance"),
+        id("tag"),
+        JSRecord("value" -> JSRecord(Nil) :: Nil)
+      ).stmt :: stmts
+    )
+    // function build(instance) { this.implement(instance); return instance; }
+    val build = JSFuncExpr(
+      S("build"),
+      param("instance") :: Nil,
+      id("this")("implement")(id("instance")).stmt
+        :: `return`(id("instance"))
+        :: Nil
+    )
+    val is = JSFuncExpr(
+      S("is"),
+      param("instance") :: Nil,
+      `return`(
+        id("instance").typeof()
+          .binary("===", JSExpr("object"))
+          .binary("&&", id("tag")
+            .binary("in", id("instance"))
+            .binary("&&", id("instance").prop(id("tag")).typeof().binary("===", JSExpr("object")))
+          )
+      ) :: Nil
+    )
+    const(
+      traitSymbol.lexicalName,
+      JSFuncExpr(
+        N,
+        Nil,
+        Ls(
+          const("tag", id("Object")()),
```

Comment:
```suggestion
          const("tag", id("Symbol")()),
```

Reply:
Didn't you want to use the `Symbol` as a key?

Reply:
Yes! I fixed it!